src/test/java/io/jenkins/plugins/folderauth/FolderAuthorizationWebAPITest.java

```diff
@@ -0,0 +1,306 @@
+                new NameValuePair("roleName", "globalRole")
+        ));
+        Page page = webClient.getPage(request);
+        assertEquals("Verifying that request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+
+        FolderBasedAuthorizationStrategy strategy = (FolderBasedAuthorizationStrategy) jenkinsRule.jenkins.getAuthorizationStrategy();
+        for (GlobalRole globalRole : strategy.getGlobalRoles()) {
+            assertNotEquals("globalRole", globalRole.getName());
+        }
+    }
+
+    @Test
+    public void testDeleteFolderRole()
+            throws IOException, NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+        testAddFolderRole();
+
+        URL apiURL = new URL(jenkinsRule.jenkins.getRootUrl() + "folder-auth/deleteFolderRole");
+        WebRequest request = new WebRequest(apiURL, HttpMethod.POST);
+
+        request.setRequestParameters(Arrays.asList(
+                new NameValuePair("roleName", "folderRole")
+        ));
+        Page page = webClient.getPage(request);
+        assertEquals("Verifying that request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+
+        FolderBasedAuthorizationStrategy strategy = (FolderBasedAuthorizationStrategy) jenkinsRule.jenkins.getAuthorizationStrategy();
+        for (FolderRole folderRole : strategy.getFolderRoles()) {
+            assertNotEquals("globalRole", folderRole.getName());
+        }
+    }
+
+    @Test
+    public void testDeleteAgentRole() throws IOException, NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+        testAddAgentRole();
+
+        URL apiURL = new URL(jenkinsRule.jenkins.getRootUrl() + "folder-auth/deleteAgentRole");
+        WebRequest request = new WebRequest(apiURL, HttpMethod.POST);
+
+        request.setRequestParameters(Arrays.asList(
+                new NameValuePair("roleName", "agentRole")
+        ));
+        Page page = webClient.getPage(request);
+        assertEquals("Verifying that request is successful", HttpURLConnection.HTTP_OK, page.getWebResponse().getStatusCode());
+
+        FolderBasedAuthorizationStrategy strategy = (FolderBasedAuthorizationStrategy) jenkinsRule.jenkins.getAuthorizationStrategy();
+        for (AgentRole agentRole : strategy.getAgentRoles()) {
+            assertNotEquals("globalRole", agentRole.getName());
+        }
+    }
+
+    private enum RoleType {
+        GLOBAL, FOLDER, AGENT
+    }
+
+    /**
+     * Util method to add a role and verify that the request succeeded
+     * @param roleType Type of role (either GLOBAL, FOLDER or AGENT
+     * @param jsonString String containing the json representation of a role
+     * @throws IOException
+     */
+    private void addRole(RoleType roleType, String jsonString) throws IOException {
+        String apiPath;
+        switch (roleType) {
+            case GLOBAL:
+                apiPath = "folder-auth/addGlobalRole";
```

Comment:
I think you can get the `folder-auth` string from https://github.com/jenkinsci/folder-auth-plugin/blob/8a9132bfc8a9ae9ec32d976b8b36347a0001fed5/src/main/java/io/jenkins/plugins/folderauth/FolderAuthorizationStrategyManagementLink.java#L76

Reply:
ok i'll refactor it to just get it from here, good practice to do so.